airflow/providers/google/common/hooks/base_google.py

```diff
@@ -670,7 +670,7 @@ async def acquire_access_token(self, timeout: int = 10) -> None:
 
         self.access_token = cast(str, self.credentials.token)
         self.access_token_duration = 3600
-        self.access_token_acquired_at = datetime.datetime.utcnow()
+        self.access_token_acquired_at = datetime.datetime.now(tz=datetime.timezone.utc)
```

Comment:
@m1racoli as original autor of this code could you tell what kind of object should be here datetime tz-naive or datetime tz-aware or doesn't matter? And I'm curious what the attribute purpouse I can't find usage in Airflow providers codebase

Reply:
Hello, we are subclassing [Token from gcloud-aio](https://github.com/talkiq/gcloud-aio/blob/auth-4.2.3/auth/gcloud/aio/auth/token.py#L164), which makes uses of those fields. A _CredentialsToken object will be used in gcloud-aio client accordingly.

Reply:
Looks like it would fail in `gcloud-aio-auth<5` if provide tz-aware datetime

https://github.com/talkiq/gcloud-aio/blob/f1132b005ba35d8059229a9ca88b90f31f77456d/auth/gcloud/aio/auth/token.py#L204

In vice versa in `gcloud-aio-auth>=5` it would fail if provide tz-naive datetime

Reply:
Or not fail. It is not straightforward to understand the order of assigning instance attributes into `gcloud-aio-auth`

Reply:
I've opened a PR.
https://github.com/apache/airflow/pull/37539